Orange/widgets/visualize/tests/test_owsilhouetteplot.py

```diff
@@ -88,3 +88,21 @@ def test_meta_object_dtype(self):
         )
         data = data.from_table(domain, data)
         self.send_signal("Data", data)
+
+    def test_memory_error(self):
+        """
+        Handling memory error.
+        GH-2336
+        """
+        from Orange.data import Table
+        import unittest
+        data = Table("iris")[::3]
+        self.assertFalse(self.widget.Error.memory_error.is_shown())
+        with unittest.mock.patch(
+            "numpy.asarray",
+            side_effect=MemoryError):
```

Comment:
So that we see the exception is caught. Which is not the case if such an exception state arises in e.g. an unguarded C code.